lib/puppet/type/cisco_vxlan_global.rb

```diff
@@ -0,0 +1,119 @@
+# Manages the VXLAN global config of a Cisco Device. It includes
+# Duplicate host IP address detection, duplicate host mac address
+# detection and configuring anycast gateway mac address.
+#
+# November 2015
+#
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_vxlan_global) do
+  @doc = "Manages the vxlan global configuration of a Cisco device.
+
+  cisco_vxlan_global { <title>:
+    ..attributes..
+  }
+
+  There can only be one instance of the cisco_vtp.
+  Example:
+    cisco_vxlan_global { default:
+      ensure   					=> present,
+      dup_host_ip_addr_detection_host_moves  	=> 200,
+      dup_host_ip_addr_detection_timeout	=> 20,
+      anycast_gateway_mac  			=> '1223.3445.5668',
+      dup_host_mac_detection_host_moves  	=> 100,
+      dup_host_mac_detection_timeout		=> 10,
+    }
+  "
+
+  newparam(:name, namevar: :true) do
+    desc "Instance of vxlan_global, only allow the value 'default'"
+    validate do |name|
+      if name != 'default'
+        error "only 'default' is accepted as a valid vxlan_global resource name"
+      end
+    end
+  end
+
+  ##############
+  # Attributes #
+  ##############
+
+  ensurable
+
+  newproperty(:dup_host_ip_addr_detection_host_moves) do
+    desc "The number of host moves allowed in n seconds. Valid values are Integer, keyword 'default'."
+
+    munge do |value|
+      begin
+        value = :default if value == 'default'
+        value = Integer(value) unless value == :default
+      rescue
+        raise 'Dup IP Addr host_moves must be an integer.'
+      end
+      value
+    end
+  end # property dup IP addr host_moves
+
+  newproperty(:dup_host_ip_addr_detection_timeout) do
+    desc "The duplicate detection timeout in seconds for the number of host moves.
+          Valid values are Integer, keyword 'default'."
+
+    munge do |value|
+      begin
+        value = :default if value == 'default'
+        value = Integer(value) unless value == :default
+      rescue
+        raise 'Dup IP Addr timeout must be an integer.'
+      end
+      value
+    end
+  end # property dup IP addr timeout
+
+  newproperty(:anycast_gateway_mac) do
+    desc 'Distributed gateway virtual MAC address. Valid values are string.'
+
+    validate do |anycast_gateway_mac|
+      fail 'Mac Address is not a string.' unless anycast_gateway_mac.is_a? String
+    end
```

Comment:
CHANGELOG: Actually it just needs to be a one liner under "New Features" like: `cisco_vxlan_global type and provider`. You can leave it like this for now and I'll fix it post-merge.

Defaults: Yes, that does work. However, consider this from a user's perspective: you clean up all of the other properties using 'default' in the manifest, but you have to remember that this particular property doesn't follow that convention and that it instead needs ' '. That seems like an unnecessary restriction for the user.